packages/integration-sdk-runtime/src/execution/duplicateKeyTracker.ts

```diff
@@ -132,8 +132,98 @@ export type DuplicateEntityReport = {
   _key: string;
   rawDataMatch: boolean;
   propertiesMatch: boolean;
+  rawDataDiff?: string;
+  propertiesDiff?: string;
 };
 
+type DiffType =
+  | 'missing_in_src'
+  | 'missing_in_dest'
+  | 'type_mismatch'
+  | 'value_mismatch';
+
+interface ObjectDiff {
+  [key: string]: {
+    type: DiffType;
+    valueTypes?: { src: string; dest: string };
+  };
+}
+
+/**
+ * Compares two objects and returns the differences between them.
+ *
+ * @param {unknown} src - The source object to compare.
+ * @param {unknown} dest - The destination object to compare.
+ * @param {string} [path=''] - The base path for keys, used for tracking nested object differences.
+ * @returns {ObjectDiff} An object representing the differences between `src` and `dest`.
+ *   Each key corresponds to a path in the objects, with details about the type of difference.
+ *
+ * @example
+ * const src = { a: 1, b: { c: 2 } };
+ * const dest = { a: 1, b: { c: 3 }, d: 4 };
+ * const result = diffObjects(src, dest);
+ * console.log(result);
+ * // Output:
+ * // {
+ * //   "b.c": { type: "value_mismatch" },
+ * //   "d": { type: "missing_in_src" }
+ * // }
+ */
+export function diffObjects(
+  src: unknown,
+  dest: unknown,
```

Comment:
nit: Source and destination are a little odd for variable names in this context.
rename options: firstEncounteredObject, secondEncounteredObject?
or a, b
or object1, object2